vm-setup/setup.sh

```diff
@@ -0,0 +1,287 @@
+        --resource-group-location) PCS_RESOURCE_GROUP_LOCATION="$2" ;;
+        --vmss-name)               PCS_VMSS_NAME="$2" ;;
+        --storage-connstring)      PCS_AZURE_STORAGE_ACCOUNT="$2" ;;
+        --iothub-name)             PCS_IOHUB_NAME="$2" ;;
+        --iothub-sku)              PCS_IOTHUB_SKU="$2" ;;
+        --iothub-tier)             PCS_IOTHUB_TIER="$2" ;;
+        --iothub-units)            PCS_IOTHUB_UNITS="$2" ;;
+        --iothub-connstring)       PCS_IOTHUB_CONNSTRING="$2" ;;
+        --app-insights-ikey)       PCS_APPINSIGHTS_INSTRUMENTATIONKEY="$2" ;;
+        --container-repository-prefix) CONTAINER_REPOSITORY_PREFIX="$2" ;;
+    esac
+    shift
+done
+
+if [ -z "$PCS_SOLUTION_SETUP_URL" ]; then
+    echo "Setup URL not specified (see --solution-setup-url)"
+    exit 1
+fi
+
+if [ -z "$PCS_RELEASE_VERSION" ]; then
+    echo "Release version not specified (see --release-version)"
+    exit 1
+fi
+
+# Note: Solution = devicesimulation
+REPOSITORY="https://raw.githubusercontent.com/Azure/azure-iot-pcs-device-simulation/${PCS_RELEASE_VERSION}/vm-setup"
+SCRIPTS_URL="${REPOSITORY}/scripts/"
+SETUP_URL="${REPOSITORY}/setup/"
+
+# ========================================================================
+
+### Install Docker and Docker Compose
+
+install_docker_ce() {
+    apt-get update -o Acquire::CompressionTypes::Order::=gz \
+        && DEBIAN_FRONTEND=noninteractive apt-get upgrade -y \
+        && apt-get update \
+        && apt-get remove docker docker-engine docker.io \
+        && apt-get -y --allow-downgrades --allow-remove-essential --allow-change-held-packages --no-install-recommends install apt-transport-https ca-certificates curl gnupg2 software-properties-common \
+        && curl -fsSL https://download.docker.com/linux/$(. /etc/os-release; echo "$ID")/gpg | sudo apt-key add - \
+        && add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/$(. /etc/os-release; echo "$ID") $(lsb_release -cs) stable" \
+        && apt-get update \
+        && DEBIAN_FRONTEND=noninteractive apt-get -y --allow-downgrades install docker-ce docker-compose \
+        && docker run --rm hello-world && docker rmi hello-world
+
+    local RESULT=$?
+    if [ $RESULT -ne 0 ]; then
+        INSTALL_DOCKER_RESULT="FAIL"
+    else
+        INSTALL_DOCKER_RESULT="OK"
+    fi
+}
+
+set +e
+INSTALL_DOCKER_RESULT="OK"
+install_docker_ce
+if [ "$INSTALL_DOCKER_RESULT" != "OK" ]; then
+    set -e
+    echo "Error: first attempt to install Docker failed, retrying..."
+    # Retry once, in case apt wasn't ready
+    sleep 30
+    install_docker_ce
+    if [ "$INSTALL_DOCKER_RESULT" != "OK" ]; then
+        echo "Error: Docker installation failed"
+        exit 1
+    fi
+fi
+set -e
+
+# ========================================================================
+
+# Configure Docker registry based on host name
+# ToDo: we may need to add similar parameter to AzureGermanCloud and AzureUSGovernment
+config_for_azure_china() {
```

Comment:
Curious why we support mooncake but none of the other sovereign clouds; was this the same case for the old SimV1?

Reply:
Unknown. This, too, is legacy. It probably made business sense at one point. I did not modify this file as part of this PR or as part of the repository consolidation.